store/storethehash/storethehash.go

```diff
@@ -181,7 +182,7 @@ func (s *sthStorage) RemoveProvider(providerID peer.ID) error {
 // DuplicateEntry checks if the entry already exists in the index. An entry
 // for the same provider but different metadata is not considered
 // a duplicate entry.
-func duplicateEntry(in store.IndexEntry, old []store.IndexEntry) bool {
+func duplicateEntry(in entry.Value, old []entry.Value) bool {
```

Comment:
Same here. We can replace `duplicateEntry` for `entry.ValueInSlice` and remove this.

Reply:
same comment as above